### PR TITLE
Fix descriptor not found error in server streaming scenario when the message is defined in another proto file

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3049
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java [hunk 1]
```diff
@@ -714,34 +714,39 @@ private String extractMessageNameWithoutNamespace(String messageName) {
...
+        byte[] annotationAsBytes = ServicesBuilderUtils.hexStringToByteArray(annotation);
+        DescriptorProtos.FileDescriptorProto file = DescriptorProtos.FileDescriptorProto.parseFrom(annotationAsBytes);
+        try {
+            Descriptors.FileDescriptor fileDescriptor = Descriptors.FileDescriptor
+                    .buildFrom(file, new Descriptors.FileDescriptor[]{}, true);
+            Descriptors.Descriptor desc = fileDescriptor.findMessageTypeByName(msgName);
+            if (desc != null) {
+                return desc;
+            }
+        } catch (Descriptors.DescriptorValidationException e) { }
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/Message.java:748]: Shall we add a comment here, explaining why we ignore the exception?

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java [hunk 2]
```diff
@@ -756,13 +761,26 @@ private String filterDescAnnotation(BMap<BString, Object> annotations) {
...
+    }
+
+    private Descriptors.Descriptor getMessageDescriptorFromMessageRegistry(String messageName) {
+        Map<String, Descriptors.Descriptor> messageDescriptorMap = MessageRegistry.getInstance()
+                .getMessageDescriptorMap();
+        if (messageDescriptorMap.containsKey(messageName)) {
+            return messageDescriptorMap.get(messageName);
+        }
+        for (String key : messageDescriptorMap.keySet().toArray(new String[]{})) { // for messages with a package name
+            if (key.endsWith("." + messageName)) {
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/Message.java:779]: this has some limitations where there are two messages with the same name and different packages. but this is not applied to the old implementations.

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java [hunk 3]
```diff
@@ -1459,42 +1477,18 @@ private String anyMessageTypeNameFromTypeUrl(String typeUrl) {
...
-    private Descriptors.Descriptor findFieldDescriptorFromTypeUrl(String messageName, RecordType recordType) {
-
-        if (messageName.startsWith("google.protobuf")) {
-            return findGoogleDescriptorFromName(messageName);
-        } else if (isDescriptorAnnotationAvailable(recordType)) {
-            try {
-                return getDescriptorFromRecord(recordType, messageName);
-            } catch (InvalidProtocolBufferException e) {
-                return MessageRegistry.getInstance().getFileDescriptor().findMessageTypeByName(messageName);
+                } catch (InvalidProtocolBufferException e) { }
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/grpc/Message.java:1491]: Please add a comment here

**Review by daneshk — APPROVED** [review]
Add minor comments, Overall LGTM

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java (hunk 3 above)
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/grpc/Message.java:1490]: Added in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/106ae179a53549ba2f9358169b4692daaf4a0c57

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java (hunk 1 above)
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/grpc/Message.java:751]: Added in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/106ae179a53549ba2f9358169b4692daaf4a0c57

File: native/src/main/java/io/ballerina/stdlib/grpc/Message.java (hunk 2 above)
Comment (dilanSachi) [reply native/src/main/java/io/ballerina/stdlib/grpc/Message.java]: Update in https://github.com/ballerina-platform/module-ballerina-grpc/pull/961/commits/8028f5629a466af37f4feb0668ae1998f4d324e5

**Pull request merged**